xpath.js

```diff
@@ -4843,5 +4899,48 @@ var xpath = (typeof exports === 'undefined') ? {} : exports;
         return exports.select(e, doc, true);
     };
 
+    var isNodeLike = function (value) {
+        return value 
+            && typeof value.nodeType === "number" 
+            && Number.isInteger(value.nodeType)
+            && value.nodeType >= 1
+            && value.nodeType <= 11
+            && typeof value.nodeName === "string";
+    };
+
+    var isArrayOfNodes = function (value) {
+        return Array.isArray(value) && value.every(isNodeLike);

```

Comment:
This will not work if any of the nodes are processing instructions.

I think we should just change `isNodeLike` to have `(typeof value.nodeName === "string" || typeof value.target === "string")`. That, along with the checks for `nodeType` should be sufficient to identify a value as nodeLike, and then we can get rid of the special casing in `isNodeOfType`.